### PR TITLE
Added more common exceptions

### PR DESCRIPTION
**Pull request by Programmer-0-0**
Added more common exceptions that will be used in #9795

### CODE REVIEW

File: app/src/main/java/eu/kanade/presentation/util/ExceptionFormatter.kt [hunk 1]
```diff
@@ -27,7 +27,44 @@ val Throwable.formattedMessage: String
...
+            "Exception" -> message ?: className
+            // General networking exceptions
+            "SocketException" -> getString(R.string.exception_socket_error)
+            "ConnectException" -> getString(R.string.exception_connect)
+            "BindException" -> getString(R.string.exception_bind_port)
+            "InterruptedIOException" -> getString(R.string.exception_io_interrupted)
+            "HttpRetryException" -> getString(R.string.exception_http_retry)
+            "PortUnreachableException" -> getString(R.string.exception_port_unreachable)
+            // General IO-related exceptions
+            "IOException" -> if (isOnline()) getString(R.string.exception_io_error) else getString(R.string.exception_offline)
```
Comment (alessandrojean) [inline app/src/main/java/eu/kanade/presentation/util/ExceptionFormatter.kt:39]: IOException is used inside the extensions interceptors to throw custom errors, so needs to be kept with the Exception guard as it was before.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/presentation/util/ExceptionFormatter.kt [hunk 2]
```diff
@@ -27,7 +27,44 @@ val Throwable.formattedMessage: String
...
+            "NoRouteToHostException" -> getString(R.string.exception_route_to_host)
+            // URL & URI related
+            "URISyntaxException" -> getString(R.string.exception_uri_syntax)
+            "MalformedURLException" -> getString(R.string.exception_malformed_url)
+            // Authentication & Proxy
+            "ProtocolException" -> getString(R.string.exception_protocol_proxy_type)
+            // Concurrency & Operation-related
+            "CancellationException" -> getString(R.string.exception_cancelled)
+            "InterruptedException" -> getString(R.string.exception_interrupted)
+            "IllegalStateException" -> getString(R.string.exception_unexpected_state)
```
Comment (alessandrojean) [inline app/src/main/java/eu/kanade/presentation/util/ExceptionFormatter.kt:64]: Some extensions also throws this with custom error messages while using `checkNotNull`, so needs to be removed as well.

**Comment by alessandrojean** [comment]
I think a good start would be addressing the fetching errors first, which are thrown by OkHttp and can't be manually thrown from inside the extensions.

**Comment by stevenyomi** [comment]
I think most of these strings are useless. Most of the exceptions are rarely seen, and explaining them doesn't enable users to troubleshoot and solve the issue themselves.

**Pull request closed without merging**